Cargo.toml

```diff
@@ -7,13 +7,21 @@ license = "Apache-2.0"
 repository = "https://github.com/oxidecomputer/diesel-dtrace.git"
 
 [dependencies]
-diesel = { git = "https://github.com/diesel-rs/diesel", rev = "ce77c382", features = [ "r2d2" ] }
+diesel = { version = "2.0.0-rc.0", features = [ "r2d2", "i-implement-a-third-party-backend-and-opt-into-breaking-changes" ] }
+# NOTE: This dependency is not directly used, and may eventually be removed.
+#
+# diesel-dtrace is frequently compiled with nightly.
+# If rust 1.61.0-nightly or older is used, version 1.4.7 breaks with the
+# following issue: https://github.com/Amanieu/parking_lot/issues/336
+#
+# As a workaround, we pin 0.4.6.
+lock_api = "=0.4.6"
```

Comment:
This is probably worth highlighting, as something to-be-removed when we don't use nightly more broadly in the stack.